From 68c69f31ecfdfa60a236ee269b13895414eaf730 Mon Sep 17 00:00:00 2001
From: mujiburrahimank <mujiburrahimank@maxlinear.com>
Date: Wed, 14 Feb 2024 15:33:25 +0530
Subject: [PATCH] WLANRTSYS-63073 : iwlwav-hostap-ng : Unable to delete last
 VAP for 6GHz Radio Band when Max VAPs are created for all Three Bands

Issue : After Creating MAX VAPS in all bands, not able to delete last created VAP in 6GHz Band, Whole interface goes down.
RCA : Buffer size to get list of interfaces for teardown/setup is less then actually needed size, So 6Ghz interfaces are not able to delete.
Fix : Optimal size is passed to ensure that 6Ghz interface's teardown/setup occurs properly and correct interface is deleted and other interfaces are up.
Signed-Off-By : Mujibur Rahiman K mujiburrahimank@maxlinear.com
---
 hostapd/ctrl_iface.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index 231ecbb1b..1408b0a97 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -7818,7 +7818,7 @@ static void hostapd_global_ctrl_iface_receive(int sock, void *eloop_ctx,
 			reply_len = -1;
 	} else if (os_strncmp(buf, "INTERFACES", 10) == 0) {
 		reply_len = hostapd_global_ctrl_iface_interfaces(
-			interfaces, buf + 10, reply, sizeof(buffer));
+			interfaces, buf + 10, reply, reply_size);
 	} else if (os_strcmp(buf, "TERMINATE") == 0) {
 		eloop_terminate();
 	} else {
-- 
2.43.0

